### PR TITLE
Status handler

### PR DESCRIPTION
**Pull request by richardmin**

### CODE REVIEW

File: handlers/status_handler.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  html_string.append("<p> Default handler: " + status.defaultHandler_ + "</p>");
+
+  html_string.append("</body>\n");
+  html_string.append("</html>\n");
+  
+
+
+  return html_string;
+}
```
Comment (avikbmohan) [inline handlers/status_handler.cc]: Not critical, but it would be good to standardize spacing here.

File: server/server.cc [hunk 2]
```diff
@@ -90,21 +101,47 @@ void Session::do_write(const std::string resp) {
...
+	if (URLCodeCount.second == false) { // The particular url is already in the map
+		URLCodeCount.first->second++;
+	}
+
+	Status_.requests_++;
+}
+
+void ServerStatus::SetDefaultHandler(std::string handler) {
+  Status_.defaultHandler_ = handler;
+}
```
Comment (avikbmohan) [inline server/server.cc]: Newline, please.

File: server/server.h [hunk 3]
```diff
@@ -33,17 +35,37 @@ class Session
   tcp::socket socket_;
   std::string data_;
   HandlerConfiguration* handler_;
+	ServerStatus* status_;
```
Comment (avikbmohan) [inline server/server.h]: I can't tell if it's github's gui but it look's like this should be un-indented

File: server/server_test.cc [hunk 4]
```diff
@@ -4,3 +4,35 @@
...
+	ASSERT_EQ(urls.size(), 2);
+
+	auto it2 = urls.begin();
+	EXPECT_EQ(it2->first, std::string("hello"));
+	EXPECT_EQ(it2->second, 1);
+	it2++;
+	EXPECT_EQ(it2->first, std::string("world"));
+	EXPECT_EQ(it2->second, 3);
+
+}
```
Comment (avikbmohan) [inline server/server_test.cc]: Newline, please.

File: server/webserver.cc [hunk 5]
```diff
@@ -23,15 +24,28 @@ bool WebServer::AddHandler(std::string path, std::string HandlerName, NginxConfi
     printf("Invalid Handler %s\n", HandlerName.c_str());
     return false;
   }
-  
+  if(HandlerName == "StatusHandler") {
+    //hacky, but I couldn't get this to work without vtable errors with dynamic casting
```
Comment (avikbmohan) [inline server/webserver.cc]: Maybe create an enum?

**Review by avikbmohan — CHANGES_REQUESTED** [review]
Please fix newline and formatting

File: server/webserver.cc (hunk 5 above)
Comment (richardmin) [reply server/webserver.cc]: it's an outdated comments whoops

**Comment by richardmin** [comment]
Updated.

**Review by avikbmohan — APPROVED** [review]

**Pull request merged**